pkg/cloudprovider/vsphere/cloud.go

```diff
@@ -68,27 +67,17 @@ func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder)
 
 		informerFactory := informers.NewSharedInformerFactory(client, controller.NoResyncPeriodFunc())
 
+		var connMgr *cm.ConnectionManager
 		if vs.cfg.Global.SecretNamespace != "" && vs.cfg.Global.SecretName != "" {
```

Comment:
Any thoughts on perhaps having this "if using K8s secrets, use the special secret Informer" within a generic `cm.NewConnectionManager()` function? That would be an added layer of abstraction, but I'm thinking about when similar logic is written in the CSI code, you then have to make a specific check to see "hey, am I running on K8s and am I using secrets", which is not something I would want to do directly (but recognize it would be happening under the covers).

Reply:
I can investigate that but will need to think about it. The issue with making that work generically for both CCM and CSI is that CCM doesnt explicitly create a k8s connection, it is passed to the CCM through the cloud provider interface. In the CSI, that connection would have to be explicitly be created.

Reply:
Yea, whether it's the entire k8s client itself or just the listener something would need to be passed in because of the CCM side. The clientbuilder object parameter on the CCM interface Initialize() wouldn't be a good candidate because there isn't one on the CSI side.

Only other option would be ditch using the clientbuilder on Initialize() on the CCM side and explicitly create a new connection in Connection Manager and then create the listener. I am not really sure what is inherited from clientbuilder when the connection is created using it.